src/adaptivemvnormal.jl

```diff
@@ -0,0 +1,84 @@
+"""
+    AdaptiveMvNormal(constant_component::MvNormal; σ=2.38, β=0.05)
+
+Adaptive multivariate normal mixture proposal as described in Haario et al. and
+Roberts & Rosenthal (2009). Uses a two-component mixture of MvNormal
+distributions. One of the components (with mixture weight `β`) remains
+constant, while the other component is adapted to the target covariance
+structure. The proposal is initialized by providing the constant component to
+the constructor. 
+
+`σ` is the scale factor for the covariance matrix, where 2.38 is supposedly
+optimal in a high-dimensional context according to Roberts & Rosenthal.
+
+# References
+
+- Haario, Heikki, Eero Saksman, and Johanna Tamminen. 
+  "An adaptive Metropolis algorithm." Bernoulli 7.2 (2001): 223-242.
+- Roberts, Gareth O., and Jeffrey S. Rosenthal. "Examples of adaptive MCMC."
+  Journal of Computational and Graphical Statistics 18.2 (2009): 349-367.
+"""
+mutable struct AdaptiveMvNormal{T1,T2,V} <: Proposal{T1}
+    d::Int  # dimensionality
+    n::Int  # iteration
+    β::Float64  # constant component mixture weight
+    σ::Float64  # scale factor for adapted covariance matrix
+    constant::T1  
+    adaptive::T2
+    Ex::Vector{V}  # rolling mean vector
+    EX::Matrix{V}  # scatter matrix of previous draws
+end
+
+function AdaptiveMvNormal(dist::MvNormal; σ=2.38, β=0.05)
+    n = length(dist)
+    adaptive = MvNormal(PDMat(Symmetric(dist.Σ)))
```

Comment:
Why is this needed? And why is it necessary to call `PDMat` and `Symmetric` explicitly? `dist.Σ` is already a `AbstractPDMat`.

Reply:
`dist.Σ` is typically a `ScalMat` (`<:AbstractPDMat`, https://github.com/JuliaStats/PDMats.jl), while the covariance matrix of `adaptive` will be a `PDMat`, so I used this to properly initialize the type for the `adaptive` field.

```
julia> typeof(p1.constant)
MvNormal{Float64,PDMats.ScalMat{Float64},FillArrays.Zeros{Float64,1,Tuple{Base.OneTo{Int64}}}}

julia> typeof(p1.adaptive)
MvNormal{Float64,PDMats.PDMat{Float64,Array{Float64,2}},FillArrays.Zeros{Float64,1,Tuple{Base.OneTo{Int64}}}}
```

Reply:
If you want to obtain a full matrix (and hence `PDMat`) for the covariance matrix of `adaptive` it should be sufficient to call `cov` (which is defined to return the full covariance matrix):
```suggestion
    adaptive = MvNormal(cov(dist))
```